SAP_Engine/Modify/OpeningType.cs

```diff
@@ -0,0 +1,170 @@
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml.Serialization;
+using System.IO;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Adapter;
+using BH.Engine.Diffing;
+using BH.oM.Environment.SAP.XML;
+using BH.oM.Environment.SAP;
+using BH.Engine.Base;
+using System.Runtime.InteropServices.ComTypes;
+using BH.oM.Environment.Elements;
+using BH.oM.Base.Attributes;
+using BH.oM.Environment.SAP.Stroma10;
+
+namespace BH.Engine.Environment.SAP
+{
+    public static partial class Modify
+    {
+        [Description("Modify opening type from a SAP report object.")]
+        [Input("sapObj", "The sap report object to modify.")]
+        [Input("type", "New type name for the windows. For each type")]
+        [Input("include", "A list of openings by name to modify.")]
+        [Input("uvalue", "Uvalue for the new type.")]
+        [Input("gvalue", "Gvalue for the new type.")]
+        [Output("sapReport", "The modified SAP Report object.")]
+        public static SAPReport ModifyOpeningTypes(this SAPReport sapObj, string type, List<string> include, double uvalue = double.NaN,  double gvalue = double.NaN)
+        {
+            if (double.IsNaN(uvalue) && double.IsNaN(gvalue))
+            {
+                return null;
+            }
+
+            if (type == null)
+            {
+                string name = string.Empty;
+
+                if (uvalue != double.NaN)
+                {
+                    name = $"uvalue_{uvalue}_{name}";
+                }
+
+                if (gvalue != double.NaN)
+                {
+                    name = name = $"gvalue_{gvalue}_{name}";
+                }
+
+                type = name;
+            }
+
+            var dict = include.Select(x => (type + "_" + x)).Zip(include, (v, k) => new { Key = k, Value = v }).ToDictionary(x => x.Key, x => x.Value);
+            var valid = dict.Select(x => (include.Contains(x.Value))).Any(x => x == true);
+
+            if (valid)
+            {
+                BH.Engine.Base.Compute.RecordError("Already a window type with these names."); //come back to this
```

Comment:
```suggestion
                BH.Engine.Base.Compute.RecordError("Already a window type with these names."); //TODO: come back to this
```

Using `TODO` in code comments can help intellisense identify areas of 'to-do' items easily 😄

Reply:
Just ask @tg359 😝